pkg/gitops/deploykey/deploykey.go

```diff
@@ -0,0 +1,155 @@
+package deploykey
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/kris-nova/logger"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+	"golang.org/x/oauth2"
+)
+
+type Manager struct {
```

Comment:
I think this file should be called `github.go` and the Manager renamed to `GitHubProvider`. To allow different integrations e.g. [GitLab API](https://docs.gitlab.com/ee/api/deploy_keys.html) we could create an interface `GitProvider` with the current Manager functions.